--- a/ovs_lib.py	2015-03-25 17:47:44.984001936 +0000
+++ b/ovs_lib.py	2015-03-25 18:04:54.647999899 +0000
@@ -36,6 +36,10 @@
     cfg.IntOpt('ovs_vsctl_timeout',
                default=DEFAULT_OVS_VSCTL_TIMEOUT,
                help=_('Timeout in seconds for ovs-vsctl commands')),
+    cfg.StrOpt('ovs_vsctl_db',
+               default='unix:/var/run/openvswitch/db.sock',
+               help=_('Sets server as the database server that ovs-vsctl '
+                      'contacts to query or modify configuration')),
 ]
 cfg.CONF.register_opts(OPTS)
 
@@ -61,10 +65,12 @@
 
     def __init__(self, root_helper):
         self.root_helper = root_helper
+        self.vsctl_db = cfg.CONF.ovs_vsctl_db
         self.vsctl_timeout = cfg.CONF.ovs_vsctl_timeout
 
     def run_vsctl(self, args, check_error=False):
-        full_args = ["ovs-vsctl", "--timeout=%d" % self.vsctl_timeout] + args
+        full_args = ["ovs-vsctl", "--db=%s" % self.vsctl_db,
+                     "--timeout=%d" % self.vsctl_timeout] + args
         try:
             return utils.execute(full_args, root_helper=self.root_helper)
         except Exception as e:
@@ -524,8 +530,8 @@
 
 
 def get_bridge_for_iface(root_helper, iface):
-    args = ["ovs-vsctl", "--timeout=%d" % cfg.CONF.ovs_vsctl_timeout,
-            "iface-to-br", iface]
+    args = ["ovs-vsctl", "--db=%s" % cfg.CONF.ovs_vsctl_db,
+            "--timeout=%d" % cfg.CONF.ovs_vsctl_timeout, "iface-to-br", iface]
     try:
         return utils.execute(args, root_helper=root_helper).strip()
     except Exception:
@@ -534,8 +540,8 @@
 
 
 def get_bridges(root_helper):
-    args = ["ovs-vsctl", "--timeout=%d" % cfg.CONF.ovs_vsctl_timeout,
-            "list-br"]
+    args = ["ovs-vsctl", "--db=%s" % cfg.CONF.ovs_vsctl_db,
+            "--timeout=%d" % cfg.CONF.ovs_vsctl_timeout, "list-br"]
     try:
         return utils.execute(args, root_helper=root_helper).strip().split("\n")
     except Exception as e:
@@ -544,8 +550,8 @@
 
 
 def get_bridge_external_bridge_id(root_helper, bridge):
-    args = ["ovs-vsctl", "--timeout=2", "br-get-external-id",
-            bridge, "bridge-id"]
+    args = ["ovs-vsctl", "--db=%s" % cfg.CONF.ovs_vsctl_db, "--timeout=2",
+            "br-get-external-id", bridge, "bridge-id"]
     try:
         return utils.execute(args, root_helper=root_helper).strip()
     except Exception:
