--- a/ovsdb_monitor.py  2015-03-26 16:51:31.997798976 +0000
+++ b/ovsdb_monitor.py  2015-03-26 16:56:52.341798889 +0000
@@ -14,9 +14,19 @@

 import eventlet

+from oslo.config import cfg
+
 from neutron.agent.linux import async_process
 from neutron.openstack.common import log as logging

+OPTS = [
+    cfg.StrOpt('ovs_vsctl_db',
+               default='unix:/var/run/openvswitch/db.sock',
+               help=_('Sets server as the database server that ovs-vsctl '
+                      'contacts to query or modify configuration')),
+]
+cfg.CONF.register_opts(OPTS)
+

 LOG = logging.getLogger(__name__)

@@ -27,7 +37,8 @@
     def __init__(self, table_name, columns=None, format=None,
                  root_helper=None, respawn_interval=None):

-        cmd = ['ovsdb-client', 'monitor', table_name]
+        cmd = ['ovsdb-client', "--db=%s" % cfg.CONF.ovs_vsctl_db, 'monitor',
+              table_name]
         if columns:
             cmd.append(','.join(columns))
         if format:
